packages/editor-ui/src/components/RunData.vue

```diff
@@ -1253,47 +1255,17 @@ export default defineComponent({
 			const itemsLabel = itemsCount > 0 ? ` (${itemsCount} ${items})` : '';
 			return option + this.$locale.baseText('ndv.output.of') + (this.maxRunIndex + 1) + itemsLabel;
 		},
-		getDataCount(
-			runIndex: number,
-			outputIndex: number,
-			connectionType: ConnectionTypes = NodeConnectionType.Main,
-		) {
-			if (this.pinData) {
-				return this.pinData.length;
-			}
-
+		getDataCount(runIndex: number) {
 			if (this.node === null) {
 				return 0;
 			}
 
 			const runData: IRunData | null = this.workflowRunData;
-
-			if (runData === null || !runData.hasOwnProperty(this.node.name)) {
-				return 0;
-			}
-
-			if (runData[this.node.name].length <= runIndex) {
-				return 0;
-			}
-
-			if (runData[this.node.name][runIndex].hasOwnProperty('error')) {
+			if (runData?.[this.node.name]?.[runIndex]?.hasOwnProperty('error')) {
 				return 1;
 			}
 
-			if (
-				!runData[this.node.name][runIndex].hasOwnProperty('data') ||
-				runData[this.node.name][runIndex].data === undefined
-			) {
-				return 0;
-			}
-
-			const inputData = this.getInputData(
-				runData[this.node.name][runIndex].data!,
-				outputIndex,
-				connectionType,
-			);
```

Comment:
Can you help me understand why we don't need to take into account the output index and connection type anymore?

Reply:
Sure, `getDataCount` method is not using anymore the `getInputData` method from `nodeHelpers` mixin, instead reusing the `inputData` computed prop (which starts with `rawInputData`)

Much of the reason for this PR was to create a single source of truth in `RunData` component for both computing the input data and prepare pagination.

Previously `inputData` used `getNodeInputData` from `nodeHelpers` mixin.
`getDataCount` used `getInputData` from `nodeHelpers` mixin to prepare pagination.
In `nodeHelpers` mixin `getNodeInputData` actually calls `getInputData`.

So to simplify this mess `getDataCount` now simply checks `inputData` length.

I know it's a lot and maybe difficult to understand in written format but we can also discuss in a call if you want

Reply:
Thank you, this makes sense 👍🏻